apps/server/src/modules/files/entity/file.entity.spec.ts

```diff
@@ -0,0 +1,309 @@
+			});
+
+			it("in case of an invalid 'deleteAt' date", () => {
+				const { file } = setup();
+
+				file.deletedAt = new Date(0);
+
+				expect(file.isMarkedForDeletion()).toEqual(false);
+			});
+
+			it("in case of the 'deleted' flag being false", () => {
+				const { file } = setup();
+
+				file.deleted = false;
+
+				expect(file.isMarkedForDeletion()).toEqual(false);
+			});
+
+			it("in case of a deleted flag being set to 'true', but no deletedAt date being set", () => {
+				const { file } = setup();
+
+				file.deleted = true;
+
+				expect(file.isMarkedForDeletion()).toEqual(false);
+			});
+		});
+	});
+
+	describe('constructor', () => {
+		describe('when creating a directory', () => {
+			it('should set proper fields values from the provided complete props object', () => {
+				const userId = new ObjectId().toHexString();
+				const props = {
+					createdAt: new Date(2023, 8, 1),
+					updatedAt: new Date(2023, 9, 1),
+					deletedAt: new Date(2023, 10, 1),
+					deleted: true,
+					isDirectory: true,
+					name: 'test-files',
+					size: 1,
+					type: 'dir',
+					storageFileName: '000-test-files',
+					bucket: '000-bucket',
+					storageProvider: storageProviderFactory.buildWithId(),
+					thumbnail: 'https://example.com/directory-thumbnail.png',
+					thumbnailRequestToken: '9d96ca2e-bc14-4fde-9a8b-948cca0bd723',
+					securityCheck: new FileSecurityCheckEntity({}),
+					shareTokens: [
+						'1c2ef176-cc1e-4e2e-bc64-0c84ad12ecb8',
+						'27ede1ff-90c1-4423-8884-a1910dc383e0',
+						'8786d3a3-7b66-431e-a19e-84a2a2f29f26',
+					],
+					parentId: new ObjectId().toHexString(),
+					ownerId: userId,
+					refOwnerModel: FileOwnerModel.USER,
+					creatorId: userId,
+					permissions: [
+						new FilePermissionEntity({
+							refId: userId,
+							refPermModel: FilePermissionReferenceModel.USER,
+						}),
+					],
+					lockId: new ObjectId().toHexString(),
+					versionKey: 0,
+				};
+
+				const entity = new FileEntity(props);
+
+				const entityProps = {
+					createdAt: entity.createdAt,
+					updatedAt: entity.updatedAt,
+					deletedAt: entity.deletedAt,
+					deleted: entity.deleted,
+					isDirectory: entity.isDirectory,
+					name: entity.name,
+					size: entity.size,
+					type: entity.type,
+					storageFileName: entity.storageFileName,
+					bucket: entity.bucket,
+					storageProvider: entity.storageProvider,
+					thumbnail: entity.thumbnail,
+					thumbnailRequestToken: entity.thumbnailRequestToken,
+					securityCheck: entity.securityCheck,
+					shareTokens: entity.shareTokens,
+					parentId: entity.parentId,
+					ownerId: entity.ownerId,
+					refOwnerModel: entity.refOwnerModel,
+					creatorId: entity.creatorId,
+					permissions: entity.permissions,
+					lockId: entity.lockId,
+					versionKey: entity.versionKey,
+				};
+
+				expect(entityProps).toEqual(props);
+			});
+		});
+
+		describe('when creating a file (non-directory)', () => {
+			const userId = new ObjectId().toHexString();
```

Comment:
@CeEv @Metauriel maybe you could make it a quick talk on one of our Thursday's meetings? This way you will also refresh the knowledge of the other teams' devs and make sure everyone is aware of the current conventions. It would be also nice to bundle it with the other rules like how to place the test consts, setup functions, maybe something on the naming conventions etc.